### PR TITLE
Add cli/sql_util and test.

### PR DESCRIPTION
**Pull request by mberhault**
Work towards #2008.
- add query runner and pretty-printer in sql_util
- add test for sql_util (including table output format)
- use placeholders rather than plain string query

### CODE REVIEW

File: cli/sql_util_test.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+func makeTestDBClient(t *testing.T, s *server.TestServer) *sql.DB {
+	db, err := sql.Open("cockroach", "https://root@"+s.ServingAddr()+"?certs=test_certs")
+	if err != nil {
+		t.Fatal(err)
+	}
+	return db
+}
+
+func TestRunQuery(t *testing.T) {
```
Comment (MBerhault) [inline cli/sql_util_test.go]: I really wanted to use the leaktest here, but the cli package imports os.signal which runs a signal loop at init time.

File: cli/sql_util.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	_ "github.com/cockroachdb/cockroach/sql/driver"
+	"github.com/cockroachdb/cockroach/util"
+	"github.com/olekukonko/tablewriter"
+)
+
+func makeSQLClient() *sql.DB {
+	// TODO(pmattis): Initialize the user to something more
+	// reasonable. Perhaps Context.Addr should be considered a URL.
+	db, err := sql.Open("cockroach",
+		fmt.Sprintf("%s://root@%s?certs=%s",
```
Comment (tamird) [inline cli/sql_util.go]: nit while you're here: use `security.RootUser` instead of "root"

Comment (MBerhault) [inline cli/sql_util.go]: done here as well as sql_util_test.go and kv.go

File: cli/sql_util_test.go (hunk 1 above)
Comment (tamird) [inline cli/sql_util_test.go]: can you update leaktest to ignore that?

File: cli/sql_util_test.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+	s := server.StartTestServer(nil)
+	db := makeTestDBClient(t, s)
+
+	defer db.Close()
+	defer s.Stop()
+
+	// Override osStdout with our own writer.
+	var b bytes.Buffer
+	osStdout = &b
+	defer func() {
```
Comment (tamird) [inline cli/sql_util_test.go]: nit: you can rewrite this such that you don't need to know the original value of `osStdout`:
```go
defer func(writer io.Writer) {
  osStdout = writer
}(osStdout)
osStdou = &b
```

**Comment by petermattis** [comment]
LGTM

File: cli/sql_util_test.go (hunk 3 above)
Comment (MBerhault) [inline cli/sql_util_test.go]: not really an issue, it's always os.Stdout.

File: cli/sql_util_test.go (hunk 1 above)
Comment (MBerhault) [inline cli/sql_util_test.go]: will take a look

File: cli/sql_util_test.go [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+		osStdout = os.Stdout
+	}()
+
+	// Non-query statement.
+	if err := runQuery(db, `SET DATABASE=system`); err != nil {
+		t.Fatal(err)
+	}
+
+	expected := `OK
```
Comment (tamird) [inline cli/sql_util_test.go]: nit: why not `"OK\n"`?

Comment (MBerhault) [inline cli/sql_util_test.go]: matches expected on the further cases.

Comment (tamird) [inline cli/sql_util_test.go]: ah :+1:

File: cli/sql_util_test.go (hunk 1 above)
Comment (MBerhault) [inline cli/sql_util_test.go]: Done.

**Comment by bdarnell** [comment]
LGTM

File: util/leaktest/leaktest.go [hunk 5]
```diff
@@ -57,7 +57,8 @@ func interestingGoroutines() (gs []string) {
 			strings.Contains(stack, "created by runtime.gc") ||
 			strings.Contains(stack, "github.com/cockroachdb/cockroach/util/leaktest.interestingGoroutines") ||
 			strings.Contains(stack, "runtime.MHeap_Scavenger") ||
-			strings.Contains(stack, "github.com/cockroachdb/cockroach/util/log.init") {
+			strings.Contains(stack, "github.com/cockroachdb/cockroach/util/log.init") ||
+			strings.Contains(stack, "os/signal.loop") {
```
Comment (tamird) [inline util/leaktest/leaktest.go]: nice.

**Comment by tamird** [comment]
LGTM

**Pull request merged**